File: Android/TUIKit/TUIRoomKit/tuiroomkit/src/main/java/com/tencent/cloud/tuikit/roomkit/ConferenceObserver.java
package com.tencent.cloud.tuikit.roomkit;

import android.util.Log;

public class ConferenceObserver {
    private static final String TAG = "ConferenceObserver";

    public void onConferenceStarted(String conferenceId, ConferenceError error) {
        Log.i(TAG, "onConferenceStarted conferenceId=" + conferenceId + " error=" + error);
    }

    public void onConferenceJoined(String conferenceId, ConferenceError error) {
        Log.i(TAG, "onConferenceJoined conferenceId=" + conferenceId + " error=" + error);
    }

    public void onConferenceExisted(String conferenceId) {
        Log.i(TAG, "onConferenceExisted conferenceId=" + conferenceId);
    }

    public void onConferenceFinished(String conferenceId) {
        Log.i(TAG, "onConferenceFinished conferenceId=" + conferenceId);
    }
}

File: iOS/TUIKit/TUIRoomKit/Source/ViewModel/ConferenceObserver.swift
//
//  ConferenceObserver.swift
//  TUIRoomKit
//
//  Created by janejntang on 2024/3/15.
//

import Foundation

@objc public protocol ConferenceObserver {
    @objc optional func onConferenceStarted(conferenceId: String, error: ConferenceError)
    @objc optional func onConferenceJoined(conferenceId: String, error: ConferenceError)
    @objc optional func onConferenceFinished(conferenceId: String)
    @objc optional func onConferenceExited(conferenceId: String)
}
